console/src/views/Radio.js

```diff
@@ -0,0 +1,193 @@
+import Button from '@enact/agate/Button';
+import Divider from '@enact/agate/Divider';
+import {LabeledItemBase} from '@enact/agate/LabeledItem';
+import {Panel} from '@enact/agate/Panels';
+import ToggleButton from '@enact/agate/ToggleButton';
+import kind from '@enact/core/kind';
+import React, {Component} from 'react';
+
+import PresetItem from '../components/PresetItem';
+
+import css from './Radio.less';
+
+
+const RadioBase = kind({
+	name: 'Radio',
+
+	handlers: {
+		onFrequencyToggle: (ev, {changeFrequency}) => {
+			const frequency = ev.currentTarget.innerText.trim();
+			changeFrequency(frequency);
+		},
+		onPresetClick: (ev, {changeStation, presets}) => {
+			changeStation(presets[ev.presetIndex]);
+		},
+		onPresetDown: (ev, {changePreset}) => {
+			changePreset(ev.presetIndex);
+		},
+		onPresetHold: (ev, {currentStation, currentPreset, updatePresets}) => {
+			updatePresets(currentStation, currentPreset);
+		},
+		onTune: (ev, {currentStation, changeStation}) => {
+			const action = ev.currentTarget.getAttribute('action');
+			let newStation;
```

Comment:
I realize this is "mock" functionality since we're not actually scanning a radio but, if it were real, I think it would be better placed in the behavior HOC than on the `kind`. The purpose and value of creating a separate `kind` is to be as pure of a representation of the translation from props to markup. As such, it's best to use handlers as adapters between the external API and the internal API and avoid "business logic" whenever possible.

In this case, I might expect an `onTune` with a `frequency` and maybe `direction` payload indicating the next station and an `onScan` event with a `direction` and the current `frequency` payload. Then it's the consumer's responsibility (HOC or app) to decide how to respond to that event.